### PR TITLE
Fix CcrRepositoryIT.testIndividualActionsTimeout()

### PR DESCRIPTION
**Pull request by tlrx**
The test `CcrRepositoryIT.testIndividualActionsTimeout()` started to exceed suite timeout (see issue #64372) after the change #61906 has been merged. In this last change we introduced an additional step to the allocation process that tries to fetch the size of the future shard to restore before assigning that shard. For CCR it means fetching the size of the store of the shard to follow, which translates to an indices stats request to the leader cluster. Like other CCR recovery actions this request uses the timeout defined by the `ccr.indices.recovery.internal_action_timeout` setting. 

But the test `CcrRepositoryIT.testIndividualActionsTimeout()` sets a low value for this setting and then tries to restore a shard, expecting that the CCR restore will fail with a timeout exception at some point.

With the change added in #61906 the indices stats request executed to fetch the shard size before allocation can now also times out, leaving the shard unassigned. When implementing #61906 we decided to not explicitly trigger reroute if such a situation occurs as we feared to trigger too many of those and add load to the cluster, expecting that such situations may resolve by themselves on real clusters (that are likely to receive more cluster state updates, or manual reroute)

Because `testIndividualActionsTimeout()` does not trigger any additional cluster updates the shard is left unassigned for too long and the whole suite times out. 

This pull request introduces a different setting to configure the timeout for fetching the shard size (which default to `ccr.indices.recovery.internal_action_timeout`). 

Another solution to force the shard to be assigned in this test would be to trigger reroute in the background, but with the same low timeout used for fetching and restoring that would mean a thread trying to assign the shard while most operations are expecting to time out.

@dnhatn @henningandersen I'm not a big fan of this workaround but I still find this ok. I'd appreciate any opinion (or ideas).

### CODE REVIEW

**Comment by henningandersen** [comment]
I think I need one more hint to understand how this fails. I thought we made it such that if fetching the shard size fails, we [proceed with allocation without size information](https://github.com/elastic/elasticsearch/pull/61906#issuecomment-704287267). Reading you description above it sounds like that is not the case here?

**Comment by tlrx** [comment]
Closed in favor of #65436
